### PR TITLE
DRILL-4971: Query encounters system error, when there aren't eval subexpressions of any function in boolean and/or expressions

### PR DESCRIPTION
**Pull request by vdiravka**
- New evaluated blocks for boolean operators should be with braces always, since they use labels.

### CODE REVIEW

File: exec/java-exec/src/main/java/org/apache/drill/exec/expr/EvaluationVisitor.java [hunk 1]
```diff
@@ -671,8 +674,9 @@ private HoldingContainer visitBooleanAnd(BooleanOperator op,
       HoldingContainer out = generator.declare(op.getMajorType());
 
       JLabel label = generator.getEvalBlockLabel("AndOP");
-      JBlock eval = generator.getEvalBlock().block();  // enter into nested block
-      generator.nestEvalBlock(eval);
+      JBlock eval = new JBlock();
```
Comment (jinfengni) [inline exec/java-exec/src/main/java/org/apache/drill/exec/expr/EvaluationVisitor.java]: How about we add the following two methods in ClassGenerator.java: 

```java
  private JBlock createInnerBlock(BlockType type) {
    final JBlock currBlock = getBlock(type);
    final JBlock innerBlock = new JBlock();
    currBlock.add(innerBlock);
    return innerBlock;
  }

  protected JBlock createInnerEvalBlock() {
    return createInnerBlock(BlockType.EVAL);
  }
```

Then, replace `generator.getEvalBlock().block()` with `generator.createInnerEvalBlock()`  ?

Comment (vdiravka) [inline exec/java-exec/src/main/java/org/apache/drill/exec/expr/EvaluationVisitor.java]: It makes sense. 
Added. Thanks.

File: exec/java-exec/src/test/java/org/apache/drill/exec/compile/TestEvaluationVisitor.java [hunk 2]
```diff
@@ -72,6 +73,37 @@ public void x() throws Exception{
     logger.debug(g.generateAndGet());
   }
 
+  @Test
+  public void testVisitBooleanOrWithoutFunctionsEvaluation() throws Exception {
```
Comment (jinfengni) [inline exec/java-exec/src/test/java/org/apache/drill/exec/compile/TestEvaluationVisitor.java]: How about move the new testcase into TestBugFixes? The new testcases uses SQL, while the existing one in TestEvaluationVisitor actually tests in a lower level.  Probably better to not mix them together.

Comment (vdiravka) [inline exec/java-exec/src/test/java/org/apache/drill/exec/compile/TestEvaluationVisitor.java]: I agree that new test cases use higher level, SQL. The first thought was TestExampleQueries class, but this class is overloaded.
TestBugFixes will be better. Thanks.

**Comment by jinfengni** [comment]
+1

**Pull request closed without merging**